Neos.Neos/Classes/Domain/Import/SiteCreationProcessor.php

```diff
@@ -0,0 +1,124 @@
+ *
+ * @phpstan-import-type DomainShape from SiteExportProcessor
+ * @phpstan-import-type SiteShape from SiteExportProcessor
+ */
+final readonly class SiteCreationProcessor implements ProcessorInterface
+{
+    public function __construct(
+        private SiteRepository $siteRepository,
+        private DomainRepository $domainRepository,
+        private PersistenceManagerInterface $persistenceManager
+    ) {
+    }
+
+    public function run(ProcessingContext $context): void
+    {
+        if ($context->files->has('sites.json')) {
+            $sitesJson = $context->files->read('sites.json');
+            try {
+                $sites = json_decode($sitesJson, true, 512, JSON_THROW_ON_ERROR);
+            } catch (JsonException $e) {
+                throw new \RuntimeException("Failed to decode sites.json: {$e->getMessage()}", 1729506117, $e);
+            }
+        } else {
+            $sites = self::extractSitesFromEventStream($context);
+        }
+
+        /** @var SiteShape $site */
+        foreach ($sites as $site) {
+            $context->dispatch(Severity::NOTICE, "Creating site \"{$site['name']}\"");
+
+            $siteNodeName = !empty($site['nodeName']) ? NodeName::fromString($site['nodeName']) : NodeName::transliterateFromString($site['name']);
+            if ($this->siteRepository->findOneByNodeName($siteNodeName->value)) {
+                $context->dispatch(Severity::NOTICE, "Site for node name \"{$siteNodeName->value}\" already exists, skipping");
+                continue;
+            }
+            // TODO use node aggregate identifier instead of node name
+            $siteInstance = new Site($siteNodeName->value);
+            $siteInstance->setSiteResourcesPackageKey($site['siteResourcesPackageKey']);
+            $siteInstance->setState(($site['online'] ?? false) ? Site::STATE_OFFLINE : Site::STATE_ONLINE);
+            $siteInstance->setName($site['name']);
+            $this->siteRepository->add($siteInstance);
+            $this->persistenceManager->persistAll();
+            foreach ($site['domains'] ?? [] as $domain) {
+                $domainInstance = new Domain();
+                $domainInstance->setSite($siteInstance);
+                $domainInstance->setHostname($domain['hostname']);
+                $domainInstance->setPort($domain['port'] ?? null);
+                $domainInstance->setScheme($domain['scheme'] ?? null);
+                $domainInstance->setActive($domain['active'] ?? false);
+                $this->domainRepository->add($domainInstance);
+                if ($domain['primary'] ?? false) {
+                    $siteInstance->setPrimaryDomain($domainInstance);
+                    $this->siteRepository->update($siteInstance);
+                }
+                $this->persistenceManager->persistAll();
+            }
+        }
+    }
+
+    /**
+     * @return array<SiteShape>
+     */
+    private static function extractSitesFromEventStream(ProcessingContext $context): array
```

Comment:
uiuiuiuiuiu but i see this is rather only a fallback in case the `sites.json` is missing .. good idea :) i would still like to mark this as `@deprecated` and also indicate on the console that were attempting a fallback strategy here:

```
$context->dispatch(Severity::WARNING, "No sites.json in export found. Attempting to decode sites from events.");
```

and then also log what sites will be created?